test/OpenTelemetry.Exporter.Geneva.Benchmark/Exporter/MetricExporterBenchmarks.cs

```diff
@@ -9,34 +9,43 @@
-| SerializeHistogramMetricItemWith4Dimensions              | 303.15 ns |  3.834 ns |  3.587 ns |         - |
-| ExportHistogramMetricItemWith3Dimensions                 | 803.90 ns | 14.753 ns | 36.465 ns |         - |
-| ExportHistogramMetricItemWith4Dimensions                 | 810.62 ns | 14.445 ns | 11.278 ns |         - |
-*/
+using OpenTelemetry.Resources;
+
+// * Summary *
+
+BenchmarkDotNet v0.13.12, Windows 11 (10.0.22631.3296/23H2/2023Update/SunValley3)
+13th Gen Intel Core i9-13900H, 1 CPU, 20 logical and 14 physical cores
+.NET SDK 8.0.202
+  [Host]     : .NET 8.0.3 (8.0.324.11423), X64 RyuJIT AVX2
+  DefaultJob : .NET 8.0.3 (8.0.324.11423), X64 RyuJIT AVX2
+
+
+| Method                                                   | Mean      | Error    | StdDev   | Gen0   | Allocated |
+|--------------------------------------------------------- |----------:|---------:|---------:|-------:|----------:|
+| InstrumentWithNoListener3Dimensions                      |  13.22 ns | 0.035 ns | 0.033 ns |      - |         - |
+| InstrumentWithNoListener4Dimensions                      |  31.12 ns | 0.386 ns | 0.361 ns |      - |         - |
+| InstrumentWithWithListener3Dimensions                    |  13.69 ns | 0.036 ns | 0.034 ns |      - |         - |
+| InstrumentWithWithListener4Dimensions                    |  30.90 ns | 0.276 ns | 0.258 ns |      - |         - |
+| InstrumentWithWithDummyReader3Dimensions                 |  66.19 ns | 0.473 ns | 0.442 ns |      - |         - |
+| InstrumentWithWithDummyReader4Dimensions                 |  94.12 ns | 1.274 ns | 1.191 ns |      - |         - |
+| InstrumentWithWithGenevaCounterMetricExporter3Dimensions |  68.57 ns | 0.489 ns | 0.457 ns |      - |         - |
+| InstrumentWithWithGenevaCounterMetricExporter4Dimensions |  94.61 ns | 0.672 ns | 0.595 ns |      - |         - |
+| SerializeCounterMetricItemWith3Dimensions                |  84.15 ns | 0.660 ns | 0.585 ns |      - |         - |
+| SerializeCounterMetricItemWith4Dimensions                |  94.97 ns | 1.129 ns | 1.056 ns |      - |         - |
+| ExportCounterMetricItemWith3Dimensions                   | 236.62 ns | 2.071 ns | 1.836 ns |      - |         - |
+| ExportCounterMetricItemWith4Dimensions                   | 254.66 ns | 2.884 ns | 2.697 ns |      - |         - |
+| SerializeHistogramMetricItemWith3Dimensions              | 142.76 ns | 2.709 ns | 2.534 ns |      - |         - |
+| SerializeHistogramMetricItemWith4Dimensions              | 156.16 ns | 2.884 ns | 2.698 ns |      - |         - |
+| ExportHistogramMetricItemWith3Dimensions                 | 329.75 ns | 3.602 ns | 3.369 ns |      - |         - |
+| ExportHistogramMetricItemWith4Dimensions                 | 325.98 ns | 6.001 ns | 5.614 ns |      - |         - |
+| SerializeCounterMetricItemWith3Dimensions_Otlp           | 227.27 ns | 3.010 ns | 2.815 ns | 0.0038 |      48 B |
```

Comment:
The allocation is coming from enumerating over resource attributes. It should go away once the [todo](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/18d616d0faa45aa92732dce9c5e7016ac329bfc4/src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs#L85) item is addressed.